[FDORA-8] feat: 판매자 대시보드 Chart  기능 구현

Bhinhb opened the pull request:
:pushpin: 작업 내용
---
- 판매자 대시보드 페이지 차트 API 구현
  - Bar, Pie형 차트
- 검색 기간 설정

<br>
<br>


:bulb: 필수 확인
---
- npm install axios (필수)

<br>
<br>


:calendar: 작업 기간
---
2025.04.12 ~ 2025.04.16

<br>
<br>


:camera: 스크린샷 (선택)
---
![screencapture-localhost-5173-my-seller-2025-04-16-18_35_54](https://github.com/user-attachments/assets/55ba03e5-54b6-49c8-a130-833b7647515f)


<br>
<br>


:white_check_mark: 참고 사항 (선택)
---

<br>
<br>

Commit:
feat: DB 연결 확인 및 Entity 변경사항 반영

- option -> options
- content type text -> longtext
- DB 연결 확인 (성공)

FDORA-8

Commit:
chore: 프로젝트 구조 세팅

- dto, controller, repository, service 생성

FDORA-8

Commit:
feat: options 테이블 수정 값 적용

FDORA-8

Commit:
feat: 일자별 매출 합계 조회

- DTO 생성
- Repository에 일자별 합계 쿼리문 작성
- 서비스 계층(?) 구현중

FDORA-8

Commit:
feat: 대시보드 차트 값 (임시)

- JPA 사용법 숙지중

FDORA-8

Commit:
feat: sellerdashboard 임시

FDORA-8

honee8583 commented on src/main/java/com/farmdora/farmdoraactivity/seller/controller/DashboardController.java:
```diff
@@ -0,0 +1,69 @@
+        response.put("labels", labels);
+        response.put("data", data);
+
+        return ResponseEntity.ok(response);
```
저희쪽에서 만든 `HttpResponse` 객체를 리턴해야 합니다.

Bhinhb replied:
이 부분 잘 이해가 안되는데 내일 설명 부탁드립니다.

honee8583 commented on src/main/java/com/farmdora/farmdoraactivity/seller/service/DashboardServiceImpl.java:
```diff
@@ -0,0 +1,38 @@
+
+    private final DashboardRepository dashboardRepository;
+
+    @Override
+    public List<SalesOverviewDTO> findDailySalesBySellerId(Integer sellerId, LocalDate startDate, LocalDate endDate) {
```
`@Transactional(readOnly = true)` 붙여주면 좋습니다.

honee8583 commented on src/main/java/com/farmdora/farmdoraactivity/seller/controller/DashboardController.java:
```diff
@@ -0,0 +1,69 @@
+        List<SalesOverviewDTO> salesData = dashboardService.findDailySalesBySellerId(sellerId, startDate, endDate);
+
+        // DTO를 차트 데이터 형식으로 변환
+        String[] labels = salesData.stream()
+                .map(sale -> sale.getCreatedDate().toString())
+                .toArray(String[]::new);
+
+        int[] data = salesData.stream()
+                .mapToInt(SalesOverviewDTO::getPrice)
+                .toArray();
+
+        Map<String, Object> response = new HashMap<>();
+        response.put("labels", labels);
+        response.put("data", data);
```
이러한 변환 로직을 서비스 쪽으로 넘기는 것도 좋을 것 같습니다.

Commit:
feat: 판매자 대시보드

- 조회 기간 설정
- 일별 매출 현황 값 표시

FDORA-8

Commit:
refactor: 코드 정리

FDORA-8

Commit:
feat: 주, 월별 매출 합계

- 주별: 시작일 (월) ~ 종료일 (일)
- 월별: 매월 1일 ~ 말일

FDORA-8

Commit:
refactor: commit 반영

FDORA-8

kjh1125 commented on src/main/java/com/farmdora/farmdoraactivity/entity/Options.java:
```diff
@@ -21,14 +21,14 @@
 @NoArgsConstructor
 @AllArgsConstructor
 @Entity
-public class Option {
```
이 부분 다시 Option으로 하기로 하고
@Table(name = "`option`") // "옆에 백팁이 있어야 되는데 댓글에 안 보이네요
public class Option {

이런 식으로 설정하기로 했습니다.
이외에도 option,like,order 등 mysql 예약어로 사용되는 것들 다 위에 @Table로 따로 지정해서 하기로 했으니 통일성을 위해서 수정하는게 좋을 거 같습니다.

Bhinhb replied:
수정 완료

kjh1125 commented on src/main/java/com/farmdora/farmdoraactivity/entity/OrderOption.java:
```diff
@@ -29,7 +29,7 @@ public class OrderOption {
     @ManyToOne(fetch = FetchType.LAZY)
     @JoinColumn(name = "option_id")
-    private Option option;
+    private Options options;
```
동일하게 Option으로 변경 다시 해야되는데 Entity쪽 이름 바꾸면 왠만하면 연관 수정되네요.

Bhinhb replied:
적용 완료

Commit:
refactor: Entity Update

FDORA-8

Commit:
feat: 제품별 판매 비율 구현

FDORA-8

Commit:
refactor: 코드 최적화

- repository, service, serviceImpl, controller DTO 최적화
- common 폴더 및 하위 파일 생성

FDORA-8

Commit:
feat: 차트 수정

- 매출이 없는 날 값 수신

FDORA-8
